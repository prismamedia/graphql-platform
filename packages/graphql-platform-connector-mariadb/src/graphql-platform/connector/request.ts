import * as mysql from 'mysql';
import { Connector } from '../connector';

export class ConnectorRequest {
  /**
   * During a "mutation", we store the connection here
   */
  public connection?: mysql.PoolConnection;

  public constructor(readonly connector: Connector) {}
}
