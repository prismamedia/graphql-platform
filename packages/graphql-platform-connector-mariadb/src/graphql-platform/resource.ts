import {
  CustomContext,
  Resource as CoreResource,
  ResourceConfig as CoreResourceConfig,
} from '@prismamedia/graphql-platform-core';
import { Maybe } from '@prismamedia/graphql-platform-utils';
import { BaseContext } from '../graphql-platform';
import { TableConfig } from './connector/database/table';
import { FieldConfig, RelationConfig } from './resource/component';
import { UniqueFullConfig } from './resource/unique';

export * from './resource/component';
export * from './resource/unique';

export interface ResourceConfig<TCustomContext extends CustomContext = any>
  extends CoreResourceConfig<
    TCustomContext,
    BaseContext,
    FieldConfig<TCustomContext>,
    RelationConfig<TCustomContext>,
    UniqueFullConfig
  > {
  table?: Maybe<TableConfig>;
}

export type Resource = CoreResource<ResourceConfig>;
