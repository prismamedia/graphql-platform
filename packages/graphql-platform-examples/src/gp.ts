import { GraphQLPlatform } from '@prismamedia/graphql-platform';
import { config as Article } from './resources/article';

export const gp = new GraphQLPlatform({
  resources: {
    Article,
  },
});

console.debug(gp.schema);
