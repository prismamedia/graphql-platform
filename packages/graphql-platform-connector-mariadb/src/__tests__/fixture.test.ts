import { Connector, Database } from '../graphql-platform/connector';
import { config, MyGP } from './gp';

describe('Fixture', () => {
  let gp: MyGP;
  let connector: Connector;
  let database: Database;

  beforeAll(async (done) => {
    gp = new MyGP(config);
    connector = gp.getConnector();
    database = connector.getDatabase();

    await database.reset();

    done();
  });

  afterAll(async (done) => {
    await database.drop();

    done();
  });

  it('loads the fixtures', async (done) => {
    // Load the fixtures without errors
    await expect(gp.loadFixtures()).resolves.toBeUndefined();

    // Query the data
    const { articles } = await gp.execute<{ articles: any[] }>({
      source: `query {
        articles(first: 5, orderBy: [slug_ASC]) {
          id
          publishedAt
          isPublished
          isImportant
          slug
          url {
            meta {
              url {
                path
              }
            }
          }
          tags(first: 5, orderBy: [order_ASC]) {
            comment {
              body
            }
            tag {
              id
              slug
              allArticleCount: articleCount
              articleWithUrlCount: articleCount(where: { article: { url_is_null: false } })
              articles(where: { article: { url_is_null: false } }, first: 10, orderBy: [createdAt_DESC]) {
                article {
                  url {
                    path
                  }
                }
              }
            }
          }
        }
      }`,
    });

    // Then check the data
    expect(articles).toMatchInlineSnapshot(`
      Array [
        Object {
          "id": "cb0456d1-d580-4d9f-bdad-510abbc96cb0",
          "isImportant": false,
          "isPublished": true,
          "publishedAt": "2018-04-30T14:20:57.004Z",
          "slug": "assumenda-ut",
          "tags": Array [
            Object {
              "comment": null,
              "tag": Object {
                "allArticleCount": 3,
                "articleWithUrlCount": 2,
                "articles": Array [
                  Object {
                    "article": Object {
                      "url": Object {
                        "path": "http://carolyne.biz",
                      },
                    },
                  },
                  Object {
                    "article": Object {
                      "url": Object {
                        "path": "https://connor.biz",
                      },
                    },
                  },
                ],
                "id": "ee8272ec-6682-4ae5-99ca-b2eb9ac55c45",
                "slug": "index-xss-baby-0",
              },
            },
            Object {
              "comment": Object {
                "body": "Est.",
              },
              "tag": Object {
                "allArticleCount": 4,
                "articleWithUrlCount": 2,
                "articles": Array [
                  Object {
                    "article": Object {
                      "url": Object {
                        "path": "http://andres.org",
                      },
                    },
                  },
                  Object {
                    "article": Object {
                      "url": Object {
                        "path": "https://connor.biz",
                      },
                    },
                  },
                ],
                "id": "7911cf17-59f6-463c-ac94-0c925980eb57",
                "slug": "calculate-bridge-upward-trending-1",
              },
            },
            Object {
              "comment": null,
              "tag": Object {
                "allArticleCount": 2,
                "articleWithUrlCount": 1,
                "articles": Array [
                  Object {
                    "article": Object {
                      "url": Object {
                        "path": "https://connor.biz",
                      },
                    },
                  },
                ],
                "id": "4838fd41-3419-49f6-850d-c43bc966dcb6",
                "slug": "platforms-proactive-3",
              },
            },
          ],
          "url": Object {
            "meta": Object {
              "url": Object {
                "path": "https://connor.biz",
              },
            },
          },
        },
        Object {
          "id": "823c9900-060a-46e9-8dfc-64f192d396a5",
          "isImportant": true,
          "isPublished": true,
          "publishedAt": "2018-04-21T05:01:06.295Z",
          "slug": "cumque-officia",
          "tags": Array [],
          "url": null,
        },
        Object {
          "id": "09b275f1-384b-4176-91da-0c6b7cb5ed46",
          "isImportant": true,
          "isPublished": false,
          "publishedAt": null,
          "slug": "dolorem-expedita",
          "tags": Array [
            Object {
              "comment": null,
              "tag": Object {
                "allArticleCount": 4,
                "articleWithUrlCount": 2,
                "articles": Array [
                  Object {
                    "article": Object {
                      "url": Object {
                        "path": "http://andres.org",
                      },
                    },
                  },
                  Object {
                    "article": Object {
                      "url": Object {
                        "path": "https://connor.biz",
                      },
                    },
                  },
                ],
                "id": "7911cf17-59f6-463c-ac94-0c925980eb57",
                "slug": "calculate-bridge-upward-trending-1",
              },
            },
          ],
          "url": Object {
            "meta": null,
          },
        },
        Object {
          "id": "d7894f73-4334-4ace-b30e-873bc4470391",
          "isImportant": true,
          "isPublished": false,
          "publishedAt": null,
          "slug": "dolorem-vel",
          "tags": Array [
            Object {
              "comment": Object {
                "body": "Illum.",
              },
              "tag": Object {
                "allArticleCount": 3,
                "articleWithUrlCount": 2,
                "articles": Array [
                  Object {
                    "article": Object {
                      "url": Object {
                        "path": "http://carolyne.biz",
                      },
                    },
                  },
                  Object {
                    "article": Object {
                      "url": Object {
                        "path": "https://connor.biz",
                      },
                    },
                  },
                ],
                "id": "ee8272ec-6682-4ae5-99ca-b2eb9ac55c45",
                "slug": "index-xss-baby-0",
              },
            },
          ],
          "url": Object {
            "meta": null,
          },
        },
        Object {
          "id": "e7431737-75b0-4383-ad06-299351e8c732",
          "isImportant": true,
          "isPublished": true,
          "publishedAt": "2018-11-09T04:29:55.177Z",
          "slug": "dolorum-soluta",
          "tags": Array [
            Object {
              "comment": null,
              "tag": Object {
                "allArticleCount": 4,
                "articleWithUrlCount": 2,
                "articles": Array [
                  Object {
                    "article": Object {
                      "url": Object {
                        "path": "http://andres.org",
                      },
                    },
                  },
                  Object {
                    "article": Object {
                      "url": Object {
                        "path": "https://connor.biz",
                      },
                    },
                  },
                ],
                "id": "7911cf17-59f6-463c-ac94-0c925980eb57",
                "slug": "calculate-bridge-upward-trending-1",
              },
            },
            Object {
              "comment": null,
              "tag": Object {
                "allArticleCount": 4,
                "articleWithUrlCount": 3,
                "articles": Array [
                  Object {
                    "article": Object {
                      "url": Object {
                        "path": "https://kiley.org",
                      },
                    },
                  },
                  Object {
                    "article": Object {
                      "url": Object {
                        "path": "https://jesse.biz",
                      },
                    },
                  },
                  Object {
                    "article": Object {
                      "url": Object {
                        "path": "https://rogelio.org",
                      },
                    },
                  },
                ],
                "id": "21e0e1c3-9b08-4a02-94f5-943142b23f95",
                "slug": "refined-soft-bike-initiative-orchid-4",
              },
            },
          ],
          "url": null,
        },
      ]
    `);

    done();
  });
});
