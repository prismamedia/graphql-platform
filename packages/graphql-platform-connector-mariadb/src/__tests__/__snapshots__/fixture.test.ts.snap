// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fixture loads the fixtures 1`] = `
"# 109 queries in 1 connections

START TRANSACTION;

INSERT INTO
  \`categories\`
SET
  \`id\` = 'dc8c0340-89bf-42ec-8468-26af53248527',
  \`title\` = 'My root category',
  \`slug\` = 'my-root-category';

COMMIT;

START TRANSACTION;

SELECT
  \`Category\`.\`id\`
FROM
  \`categories\` AS \`Category\`
WHERE
  \`Category\`.\`id\` = 'dc8c0340-89bf-42ec-8468-26af53248527'
LIMIT
  1;

INSERT INTO
  \`categories\`
SET
  \`id\` = '00dd8125-a01e-4a96-9e56-35f3acfec7d1',
  \`title\` = 'My first category',
  \`slug\` = 'my-first-category',
  \`parentId\` = 'dc8c0340-89bf-42ec-8468-26af53248527';

COMMIT;

START TRANSACTION;

INSERT INTO
  \`users\`
SET
  \`id\` = 'fdb371f6-3b17-427c-8a46-08b6368312b5',
  \`username\` = 'A publisher';

COMMIT;

START TRANSACTION;

SELECT
  \`Category\`.\`parentId\`,
  \`Category\`.\`slug\`
FROM
  \`categories\` AS \`Category\`
WHERE
  \`Category\`.\`id\` = '00dd8125-a01e-4a96-9e56-35f3acfec7d1'
LIMIT
  1;

SELECT
  \`User\`.\`username\`
FROM
  \`users\` AS \`User\`
WHERE
  \`User\`.\`id\` = 'fdb371f6-3b17-427c-8a46-08b6368312b5'
LIMIT
  1;

SELECT
  MAX(\`intId\`) as maxIntId
FROM
  \`articles\`;

SELECT
  \`Article\`.\`slug\`
FROM
  \`articles\` AS \`Article\`
WHERE
  (\`Article\`.\`format\` IN ('RICH', 'VIDEO'))
  AND (
    (\`Article\`.\`categoryParentId\` IS NULL)
    AND (
      \`Article\`.\`categorySlug\` = 'a-non-existent-category-slug'
    )
  )
  AND (\`Article\`.\`slug\` = 'a-non-existent-article-slug')
LIMIT
  1;

INSERT INTO
  \`articles\`
SET
  \`id\` = '9b6b98fa-586d-4987-82a0-d63f11bbe560',
  \`computedMaxIntId\` = 100,
  \`publishedAt\` = '2019-07-01T03:41:37.829',
  \`format\` = 'RICH',
  \`title\` = 'My first article\\\\'s title, a rich article.',
  \`slug\` = 'my-first-articles-title-a-rich-article',
  \`categoryParentId\` = 'dc8c0340-89bf-42ec-8468-26af53248527',
  \`categorySlug\` = 'my-first-category',
  \`authorUsername\` = 'A publisher';

SELECT
  \`Article\`.\`id\`,
  \`Article\`.\`intId\`,
  \`Article\`.\`createdAt\`,
  \`Article\`.\`updatedAt\`,
  \`Article\`.\`computedMaxIntId\`,
  \`Article\`.\`publishedAt\`,
  \`Article\`.\`format\`,
  \`Article\`.\`title\`,
  \`Article\`.\`slug\`,
  \`Article\`.\`body\`,
  \`Article\`.\`categoryParentId\`,
  \`Article\`.\`categorySlug\`,
  \`Article>category\`.\`parentId\`,
  \`Article\`.\`authorUsername\`,
  \`Article\`.\`moderatorIntId\`
FROM
  \`articles\` AS \`Article\`
  LEFT JOIN \`categories\` AS \`Article>category\` ON (
    \`Article\`.\`categoryParentId\` = \`Article>category\`.\`parentId\`
  )
  AND (
    \`Article\`.\`categorySlug\` = \`Article>category\`.\`slug\`
  )
WHERE
  (\`Article\`.\`format\` IN ('RICH', 'VIDEO'))
  AND (\`Article\`.\`intId\` = 1)
LIMIT
  1;

COMMIT;

START TRANSACTION;

INSERT INTO
  \`tags\`
SET
  \`id\` = '3480abaa-5580-4dbe-8e55-cb0b622e1da5',
  \`title\` = 'My third tag',
  \`slug\` = 'my-third-tag';

COMMIT;

START TRANSACTION;

SELECT
  \`Article\`.\`intId\`
FROM
  \`articles\` AS \`Article\`
WHERE
  (\`Article\`.\`format\` IN ('RICH', 'VIDEO'))
  AND (
    \`Article\`.\`id\` = '9b6b98fa-586d-4987-82a0-d63f11bbe560'
  )
LIMIT
  1;

SELECT
  \`Tag\`.\`intId\`
FROM
  \`tags\` AS \`Tag\`
WHERE
  \`Tag\`.\`id\` = '3480abaa-5580-4dbe-8e55-cb0b622e1da5'
LIMIT
  1;

INSERT INTO
  \`article_tags\`
SET
  \`articleIntId\` = 1,
  \`tagIntId\` = 1,
  \`order\` = 2;

SELECT
  \`ArticleTag>article\`.\`id\`,
  \`ArticleTag>tag\`.\`id\`
FROM
  \`article_tags\` AS \`ArticleTag\`
  LEFT JOIN \`articles\` AS \`ArticleTag>article\` ON \`ArticleTag\`.\`articleIntId\` = \`ArticleTag>article\`.\`intId\`
  LEFT JOIN \`tags\` AS \`ArticleTag>tag\` ON \`ArticleTag\`.\`tagIntId\` = \`ArticleTag>tag\`.\`intId\`
WHERE
  (\`ArticleTag\`.\`articleIntId\` = 1)
  AND (\`ArticleTag\`.\`tagIntId\` = 1)
LIMIT
  1;

COMMIT;

START TRANSACTION;

INSERT INTO
  \`tags\`
SET
  \`id\` = '1b4cc2fd-0e1f-4b97-bed5-171f52d18b1b',
  \`title\` = 'My fourth tag',
  \`slug\` = 'my-fourth-tag';

COMMIT;

START TRANSACTION;

SELECT
  \`Article\`.\`intId\`
FROM
  \`articles\` AS \`Article\`
WHERE
  (\`Article\`.\`format\` IN ('RICH', 'VIDEO'))
  AND (
    \`Article\`.\`id\` = '9b6b98fa-586d-4987-82a0-d63f11bbe560'
  )
LIMIT
  1;

SELECT
  \`Tag\`.\`intId\`
FROM
  \`tags\` AS \`Tag\`
WHERE
  \`Tag\`.\`id\` = '1b4cc2fd-0e1f-4b97-bed5-171f52d18b1b'
LIMIT
  1;

INSERT INTO
  \`article_tags\`
SET
  \`articleIntId\` = 1,
  \`tagIntId\` = 2,
  \`order\` = 3;

SELECT
  \`ArticleTag>article\`.\`id\`,
  \`ArticleTag>tag\`.\`id\`
FROM
  \`article_tags\` AS \`ArticleTag\`
  LEFT JOIN \`articles\` AS \`ArticleTag>article\` ON \`ArticleTag\`.\`articleIntId\` = \`ArticleTag>article\`.\`intId\`
  LEFT JOIN \`tags\` AS \`ArticleTag>tag\` ON \`ArticleTag\`.\`tagIntId\` = \`ArticleTag>tag\`.\`intId\`
WHERE
  (\`ArticleTag\`.\`articleIntId\` = 1)
  AND (\`ArticleTag\`.\`tagIntId\` = 2)
LIMIT
  1;

COMMIT;

START TRANSACTION;

SELECT
  \`Category\`.\`id\`
FROM
  \`categories\` AS \`Category\`
WHERE
  \`Category\`.\`id\` = '00dd8125-a01e-4a96-9e56-35f3acfec7d1'
LIMIT
  1;

INSERT INTO
  \`categories\`
SET
  \`id\` = '947b872e-0a54-406d-850b-1aeae571463b',
  \`title\` = 'My sub-category in my first category',
  \`slug\` = 'my-sub-category-in-my-first-category',
  \`parentId\` = '00dd8125-a01e-4a96-9e56-35f3acfec7d1';

COMMIT;

START TRANSACTION;

INSERT INTO
  \`users\`
SET
  \`id\` = '980bca6a-c034-4841-ab3e-4b636eff7be0',
  \`username\` = 'A moderator';

COMMIT;

START TRANSACTION;

SELECT
  \`Category\`.\`parentId\`,
  \`Category\`.\`slug\`
FROM
  \`categories\` AS \`Category\`
WHERE
  \`Category\`.\`id\` = '947b872e-0a54-406d-850b-1aeae571463b'
LIMIT
  1;

SELECT
  \`User\`.\`username\`
FROM
  \`users\` AS \`User\`
WHERE
  \`User\`.\`id\` = 'fdb371f6-3b17-427c-8a46-08b6368312b5'
LIMIT
  1;

SELECT
  \`User\`.\`intId\`
FROM
  \`users\` AS \`User\`
WHERE
  \`User\`.\`id\` = '980bca6a-c034-4841-ab3e-4b636eff7be0'
LIMIT
  1;

SELECT
  MAX(\`intId\`) as maxIntId
FROM
  \`articles\`;

SELECT
  \`Article\`.\`slug\`
FROM
  \`articles\` AS \`Article\`
WHERE
  (\`Article\`.\`format\` IN ('RICH', 'VIDEO'))
  AND (
    (\`Article\`.\`categoryParentId\` IS NULL)
    AND (
      \`Article\`.\`categorySlug\` = 'a-non-existent-category-slug'
    )
  )
  AND (\`Article\`.\`slug\` = 'a-non-existent-article-slug')
LIMIT
  1;

INSERT INTO
  \`articles\`
SET
  \`id\` = '06bd57da-24f6-488a-a1b1-2435bd7c8d6e',
  \`computedMaxIntId\` = 101,
  \`format\` = 'VIDEO',
  \`title\` = 'My second article\\\\'s title, a video.',
  \`slug\` = 'my-second-articles-title-a-video',
  \`categoryParentId\` = '00dd8125-a01e-4a96-9e56-35f3acfec7d1',
  \`categorySlug\` = 'my-sub-category-in-my-first-category',
  \`authorUsername\` = 'A publisher',
  \`moderatorIntId\` = 2;

SELECT
  \`Article\`.\`id\`,
  \`Article\`.\`intId\`,
  \`Article\`.\`createdAt\`,
  \`Article\`.\`updatedAt\`,
  \`Article\`.\`computedMaxIntId\`,
  \`Article\`.\`publishedAt\`,
  \`Article\`.\`format\`,
  \`Article\`.\`title\`,
  \`Article\`.\`slug\`,
  \`Article\`.\`body\`,
  \`Article\`.\`categoryParentId\`,
  \`Article\`.\`categorySlug\`,
  \`Article>category\`.\`parentId\`,
  \`Article\`.\`authorUsername\`,
  \`Article\`.\`moderatorIntId\`
FROM
  \`articles\` AS \`Article\`
  LEFT JOIN \`categories\` AS \`Article>category\` ON (
    \`Article\`.\`categoryParentId\` = \`Article>category\`.\`parentId\`
  )
  AND (
    \`Article\`.\`categorySlug\` = \`Article>category\`.\`slug\`
  )
WHERE
  (\`Article\`.\`format\` IN ('RICH', 'VIDEO'))
  AND (\`Article\`.\`intId\` = 2)
LIMIT
  1;

COMMIT;

START TRANSACTION;

SELECT
  \`Article\`.\`intId\`
FROM
  \`articles\` AS \`Article\`
WHERE
  (\`Article\`.\`format\` IN ('RICH', 'VIDEO'))
  AND (
    \`Article\`.\`id\` = '06bd57da-24f6-488a-a1b1-2435bd7c8d6e'
  )
LIMIT
  1;

SELECT
  \`Tag\`.\`intId\`
FROM
  \`tags\` AS \`Tag\`
WHERE
  \`Tag\`.\`id\` = '1b4cc2fd-0e1f-4b97-bed5-171f52d18b1b'
LIMIT
  1;

INSERT INTO
  \`article_tags\`
SET
  \`articleIntId\` = 2,
  \`tagIntId\` = 2,
  \`order\` = 0;

SELECT
  \`ArticleTag>article\`.\`id\`,
  \`ArticleTag>tag\`.\`id\`
FROM
  \`article_tags\` AS \`ArticleTag\`
  LEFT JOIN \`articles\` AS \`ArticleTag>article\` ON \`ArticleTag\`.\`articleIntId\` = \`ArticleTag>article\`.\`intId\`
  LEFT JOIN \`tags\` AS \`ArticleTag>tag\` ON \`ArticleTag\`.\`tagIntId\` = \`ArticleTag>tag\`.\`intId\`
WHERE
  (\`ArticleTag\`.\`articleIntId\` = 2)
  AND (\`ArticleTag\`.\`tagIntId\` = 2)
LIMIT
  1;

COMMIT;

START TRANSACTION;

INSERT INTO
  \`tags\`
SET
  \`id\` = '91eef28c-598a-433c-9e2c-06d6bd6b455c',
  \`title\` = 'My fifth tag',
  \`slug\` = 'my-fifth-tag';

COMMIT;

START TRANSACTION;

SELECT
  \`Article\`.\`intId\`
FROM
  \`articles\` AS \`Article\`
WHERE
  (\`Article\`.\`format\` IN ('RICH', 'VIDEO'))
  AND (
    \`Article\`.\`id\` = '06bd57da-24f6-488a-a1b1-2435bd7c8d6e'
  )
LIMIT
  1;

SELECT
  \`Tag\`.\`intId\`
FROM
  \`tags\` AS \`Tag\`
WHERE
  \`Tag\`.\`id\` = '91eef28c-598a-433c-9e2c-06d6bd6b455c'
LIMIT
  1;

INSERT INTO
  \`article_tags\`
SET
  \`articleIntId\` = 2,
  \`tagIntId\` = 3,
  \`order\` = 1;

SELECT
  \`ArticleTag>article\`.\`id\`,
  \`ArticleTag>tag\`.\`id\`
FROM
  \`article_tags\` AS \`ArticleTag\`
  LEFT JOIN \`articles\` AS \`ArticleTag>article\` ON \`ArticleTag\`.\`articleIntId\` = \`ArticleTag>article\`.\`intId\`
  LEFT JOIN \`tags\` AS \`ArticleTag>tag\` ON \`ArticleTag\`.\`tagIntId\` = \`ArticleTag>tag\`.\`intId\`
WHERE
  (\`ArticleTag\`.\`articleIntId\` = 2)
  AND (\`ArticleTag\`.\`tagIntId\` = 3)
LIMIT
  1;

COMMIT;

START TRANSACTION;

INSERT INTO
  \`tags\`
SET
  \`id\` = 'd0ae7305-ddce-4870-a0f9-0f7ef19b86ea',
  \`title\` = 'My first tag',
  \`slug\` = 'my-first-tag';

COMMIT;

START TRANSACTION;

SELECT
  \`Article\`.\`intId\`
FROM
  \`articles\` AS \`Article\`
WHERE
  (\`Article\`.\`format\` IN ('RICH', 'VIDEO'))
  AND (
    \`Article\`.\`id\` = '9b6b98fa-586d-4987-82a0-d63f11bbe560'
  )
LIMIT
  1;

SELECT
  \`Tag\`.\`intId\`
FROM
  \`tags\` AS \`Tag\`
WHERE
  \`Tag\`.\`id\` = 'd0ae7305-ddce-4870-a0f9-0f7ef19b86ea'
LIMIT
  1;

INSERT INTO
  \`article_tags\`
SET
  \`articleIntId\` = 1,
  \`tagIntId\` = 4,
  \`order\` = 0;

SELECT
  \`ArticleTag>article\`.\`id\`,
  \`ArticleTag>tag\`.\`id\`
FROM
  \`article_tags\` AS \`ArticleTag\`
  LEFT JOIN \`articles\` AS \`ArticleTag>article\` ON \`ArticleTag\`.\`articleIntId\` = \`ArticleTag>article\`.\`intId\`
  LEFT JOIN \`tags\` AS \`ArticleTag>tag\` ON \`ArticleTag\`.\`tagIntId\` = \`ArticleTag>tag\`.\`intId\`
WHERE
  (\`ArticleTag\`.\`articleIntId\` = 1)
  AND (\`ArticleTag\`.\`tagIntId\` = 4)
LIMIT
  1;

COMMIT;

START TRANSACTION;

SELECT
  \`ArticleTag\`.\`articleIntId\`,
  \`ArticleTag\`.\`tagIntId\`
FROM
  \`article_tags\` AS \`ArticleTag\`
  LEFT JOIN \`articles\` AS \`ArticleTag>article\` ON \`ArticleTag\`.\`articleIntId\` = \`ArticleTag>article\`.\`intId\`
  LEFT JOIN \`tags\` AS \`ArticleTag>tag\` ON \`ArticleTag\`.\`tagIntId\` = \`ArticleTag>tag\`.\`intId\`
WHERE
  (
    \`ArticleTag>article\`.\`id\` = '9b6b98fa-586d-4987-82a0-d63f11bbe560'
  )
  AND (
    \`ArticleTag>tag\`.\`id\` = 'd0ae7305-ddce-4870-a0f9-0f7ef19b86ea'
  )
LIMIT
  1;

INSERT INTO
  \`article_tag_comments\`
SET
  \`articleTagArticleIntId\` = 1,
  \`articleTagTagIntId\` = 4,
  \`body\` = 'This is why I tagged the Article \\\\\\"article_01\\\\\\" with the Tag \\\\\\"tag_01\\\\\\"';

SELECT
  \`ArticleTagComment>articleTag>article\`.\`id\`,
  \`ArticleTagComment>articleTag>tag\`.\`id\`
FROM
  \`article_tag_comments\` AS \`ArticleTagComment\`
  LEFT JOIN \`article_tags\` AS \`ArticleTagComment>articleTag\` ON (
    \`ArticleTagComment\`.\`articleTagArticleIntId\` = \`ArticleTagComment>articleTag\`.\`articleIntId\`
  )
  AND (
    \`ArticleTagComment\`.\`articleTagTagIntId\` = \`ArticleTagComment>articleTag\`.\`tagIntId\`
  )
  LEFT JOIN \`articles\` AS \`ArticleTagComment>articleTag>article\` ON \`ArticleTagComment>articleTag\`.\`articleIntId\` = \`ArticleTagComment>articleTag>article\`.\`intId\`
  LEFT JOIN \`tags\` AS \`ArticleTagComment>articleTag>tag\` ON \`ArticleTagComment>articleTag\`.\`tagIntId\` = \`ArticleTagComment>articleTag>tag\`.\`intId\`
WHERE
  (\`ArticleTagComment\`.\`articleTagArticleIntId\` = 1)
  AND (\`ArticleTagComment\`.\`articleTagTagIntId\` = 4)
LIMIT
  1;

COMMIT;

START TRANSACTION;

INSERT INTO
  \`tags\`
SET
  \`id\` = '511ed2c7-0557-4e5d-8bce-d4956cac48eb',
  \`title\` = 'My second tag',
  \`slug\` = 'my-second-tag';

COMMIT;

START TRANSACTION;

SELECT
  \`Article\`.\`intId\`
FROM
  \`articles\` AS \`Article\`
WHERE
  (\`Article\`.\`format\` IN ('RICH', 'VIDEO'))
  AND (
    \`Article\`.\`id\` = '9b6b98fa-586d-4987-82a0-d63f11bbe560'
  )
LIMIT
  1;

SELECT
  \`Tag\`.\`intId\`
FROM
  \`tags\` AS \`Tag\`
WHERE
  \`Tag\`.\`id\` = '511ed2c7-0557-4e5d-8bce-d4956cac48eb'
LIMIT
  1;

INSERT INTO
  \`article_tags\`
SET
  \`articleIntId\` = 1,
  \`tagIntId\` = 5,
  \`order\` = 1;

SELECT
  \`ArticleTag>article\`.\`id\`,
  \`ArticleTag>tag\`.\`id\`
FROM
  \`article_tags\` AS \`ArticleTag\`
  LEFT JOIN \`articles\` AS \`ArticleTag>article\` ON \`ArticleTag\`.\`articleIntId\` = \`ArticleTag>article\`.\`intId\`
  LEFT JOIN \`tags\` AS \`ArticleTag>tag\` ON \`ArticleTag\`.\`tagIntId\` = \`ArticleTag>tag\`.\`intId\`
WHERE
  (\`ArticleTag\`.\`articleIntId\` = 1)
  AND (\`ArticleTag\`.\`tagIntId\` = 5)
LIMIT
  1;

COMMIT;

START TRANSACTION;

SELECT
  \`ArticleTag\`.\`articleIntId\`,
  \`ArticleTag\`.\`tagIntId\`
FROM
  \`article_tags\` AS \`ArticleTag\`
  LEFT JOIN \`articles\` AS \`ArticleTag>article\` ON \`ArticleTag\`.\`articleIntId\` = \`ArticleTag>article\`.\`intId\`
  LEFT JOIN \`tags\` AS \`ArticleTag>tag\` ON \`ArticleTag\`.\`tagIntId\` = \`ArticleTag>tag\`.\`intId\`
WHERE
  (
    \`ArticleTag>article\`.\`id\` = '9b6b98fa-586d-4987-82a0-d63f11bbe560'
  )
  AND (
    \`ArticleTag>tag\`.\`id\` = '511ed2c7-0557-4e5d-8bce-d4956cac48eb'
  )
LIMIT
  1;

INSERT INTO
  \`article_tag_comments\`
SET
  \`articleTagArticleIntId\` = 1,
  \`articleTagTagIntId\` = 5,
  \`body\` = 'And this is why I tagged the Article \\\\\\"article_01\\\\\\" with the Tag \\\\\\"tag_02\\\\\\"';

SELECT
  \`ArticleTagComment>articleTag>article\`.\`id\`,
  \`ArticleTagComment>articleTag>tag\`.\`id\`
FROM
  \`article_tag_comments\` AS \`ArticleTagComment\`
  LEFT JOIN \`article_tags\` AS \`ArticleTagComment>articleTag\` ON (
    \`ArticleTagComment\`.\`articleTagArticleIntId\` = \`ArticleTagComment>articleTag\`.\`articleIntId\`
  )
  AND (
    \`ArticleTagComment\`.\`articleTagTagIntId\` = \`ArticleTagComment>articleTag\`.\`tagIntId\`
  )
  LEFT JOIN \`articles\` AS \`ArticleTagComment>articleTag>article\` ON \`ArticleTagComment>articleTag\`.\`articleIntId\` = \`ArticleTagComment>articleTag>article\`.\`intId\`
  LEFT JOIN \`tags\` AS \`ArticleTagComment>articleTag>tag\` ON \`ArticleTagComment>articleTag\`.\`tagIntId\` = \`ArticleTagComment>articleTag>tag\`.\`intId\`
WHERE
  (\`ArticleTagComment\`.\`articleTagArticleIntId\` = 1)
  AND (\`ArticleTagComment\`.\`articleTagTagIntId\` = 5)
LIMIT
  1;

COMMIT;

START TRANSACTION;

SELECT
  \`Article\`.\`intId\`
FROM
  \`articles\` AS \`Article\`
WHERE
  (\`Article\`.\`format\` IN ('RICH', 'VIDEO'))
  AND (
    \`Article\`.\`id\` = '9b6b98fa-586d-4987-82a0-d63f11bbe560'
  )
LIMIT
  1;

INSERT INTO
  \`article_urls\`
SET
  \`articleIntId\` = 1,
  \`path\` = '/my-article-01';

SELECT
  \`ArticleUrl>article\`.\`id\`
FROM
  \`article_urls\` AS \`ArticleUrl\`
  LEFT JOIN \`articles\` AS \`ArticleUrl>article\` ON \`ArticleUrl\`.\`articleIntId\` = \`ArticleUrl>article\`.\`intId\`
WHERE
  \`ArticleUrl\`.\`articleIntId\` = 1
LIMIT
  1;

COMMIT;

START TRANSACTION;

SELECT
  \`ArticleUrl\`.\`articleIntId\`
FROM
  \`article_urls\` AS \`ArticleUrl\`
  LEFT JOIN \`articles\` AS \`ArticleUrl>article\` ON \`ArticleUrl\`.\`articleIntId\` = \`ArticleUrl>article\`.\`intId\`
WHERE
  \`ArticleUrl>article\`.\`id\` = '9b6b98fa-586d-4987-82a0-d63f11bbe560'
LIMIT
  1;

INSERT INTO
  \`article_url_metas\`
SET
  \`urlArticleIntId\` = 1;

SELECT
  \`ArticleUrlMeta>url>article\`.\`id\`
FROM
  \`article_url_metas\` AS \`ArticleUrlMeta\`
  LEFT JOIN \`article_urls\` AS \`ArticleUrlMeta>url\` ON \`ArticleUrlMeta\`.\`urlArticleIntId\` = \`ArticleUrlMeta>url\`.\`articleIntId\`
  LEFT JOIN \`articles\` AS \`ArticleUrlMeta>url>article\` ON \`ArticleUrlMeta>url\`.\`articleIntId\` = \`ArticleUrlMeta>url>article\`.\`intId\`
WHERE
  \`ArticleUrlMeta\`.\`urlArticleIntId\` = 1
LIMIT
  1;

COMMIT;

START TRANSACTION;

SELECT
  \`Category\`.\`id\`
FROM
  \`categories\` AS \`Category\`
WHERE
  \`Category\`.\`id\` = 'dc8c0340-89bf-42ec-8468-26af53248527'
LIMIT
  1;

INSERT INTO
  \`categories\`
SET
  \`id\` = 'f45c709f-96ca-40cd-afe7-ed436406db03',
  \`title\` = 'My second category',
  \`slug\` = 'my-second-category',
  \`parentId\` = 'dc8c0340-89bf-42ec-8468-26af53248527';

COMMIT;"
`;
